src/t_hash.c

```diff
@@ -772,7 +773,15 @@ GetFieldRes hashTypeGetValue(redisDb *db, robj *o, sds field, unsigned char **vs
             notifyKeyspaceEvent(NOTIFY_GENERIC, "del", keyObj, db->id);
         dbDelete(db,keyObj);
         res = GETF_EXPIRED_HASH;
-    }
+    } else {
+        /* Check if it is the removed HFE is the last one in the hash, then we must
+         * remove the hash from global HFE for the sake of subexpiry statistics. In
+         * case there are more HFEs, then no need to update. Active expiration
+         * will update the minimum expiration time of the hash in the global HFE
+         * DS gracefully (Optimization). */
```

Comment:
Without this fix, subexpiry stats will give wrong number until active expiration kicks in, right? This is what we are trying to fix here? Trying to be more accurate? If so, maybe you can make it clear here or at least in the top comment of the PR.